### PR TITLE
Jepsen/mdbook

### PR DESCRIPTION
**Pull request by 0xJepsen**
I added some more to the documentation here

- [x] Events
- [x] Subscriptions

contributes to #2282

### CODE REVIEW

File: book/events/events.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+3. SubscriptionStream: A stream of events you can subscribe to for real-time updates.
+
+## Getting Started
+
+Before diving into event handling, ensure you have ethers-rs added to your project's dependencies in Cargo.toml:
+
+```toml
+[dependencies]
+ethers = { version = "2.0.0.", features = ["full"] }
+Now, let's import the necessary components from the ethers-rs library:
```
Comment (prestwich) [inline book/events/events.md:20]: Should be outside the code block?

File: book/events/events.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+}
+```
+
+First, define a struct representing the ValueChanged event:
+
+```rust
+#[derive(Debug, Clone, EthEvent)]
+pub struct ValueChanged {
+pub new_value: U256,
```
Comment (prestwich) [inline book/events/events.md:57]: indent?

File: book/events/events.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+```toml
+[dependencies]
+ethers = { version = "2.0.0.", features = ["full"] }
+Now, let's import the necessary components from the ethers-rs library:
+```
+
+```rust
+use ethers::{
+prelude::\*,
```
Comment (prestwich) [inline book/events/events.md:25]: indent. is the escape on the asterisk necessary?

File: book/subscriptions/subscriptions.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+```toml
+[dependencies]
+ethers = { version = "2.0.0", features = ["full"] }
+```
+
+Next, import the necessary components from the ethers-rs library:
+
+```rust
+use ethers::{prelude::\*,types::H256,};
```
Comment (prestwich) [inline book/subscriptions/subscriptions.md:24]: is the escape necessary here?

**Review by prestwich — CHANGES_REQUESTED** [review]
High-level changes:

It would be good to route users to the `abigen!`-based interfaces for accessing these tools, as they're simpler and higher level. This documentation is a good explanation of how abigen works under the hood

We should also mention the difference between `Event::stream` and `Event::subscribe`. the `stream` interface is polling, and will work with any provider, but have increased cost and latency. Subscribe is based on WS/IPC notifications, and is limited to those providers

File: book/events/events.md (hunk 1 above)
Comment (0xJepsen) [reply book/events/events.md:20]: nice catch

**Comment by 0xJepsen** [comment]
Thanks for the feedback, I went and made some of the easy changes. I have yet to distinguish the difference between `Event::stream` and `Event::subscribe` like you recommended but will look to getting around to it soon.

**Comment by prestwich** [comment]
> Thanks for the feedback, I went and made some of the easy changes. I have yet to distinguish the difference between Event::stream and Event::subscribe like you recommended but will look to getting around to it soon.

did you intend to do this as part of this PR? or in a future one?

**Comment by prestwich** [comment]
CI failures unrelated

**Pull request merged**